Doc/library/re.rst

```diff
@@ -1562,21 +1562,24 @@ search() vs. match()
 
 .. sectionauthor:: Fred L. Drake, Jr. <fdrake@acm.org>
 
-Python offers two different primitive operations based on regular expressions:
-:func:`re.match` checks for a match only at the beginning of the string, while
-:func:`re.search` checks for a match anywhere in the string (this is what Perl
-does by default).
+Python offers different primitive operations based on regular expressions:
+:func:`re.match` checks for a match only at the beginning of the string,
+:func:`re.fullmatch` checks for entire string to be a match. :func:`re.search`
+checks for a match anywhere in the string (this is what Perl does by default).
```

Comment:
For better flow

```suggestion
:func:`re.fullmatch` checks for entire string to be a match, and
:func:`re.search` checks for a match anywhere in the string
(this is what Perl does by default).
```

or perhaps even

```suggestion
:func:`re.fullmatch` checks for entire string to be a match, and
:func:`re.search` checks for a match anywhere in the string.
```

but I'll let Eric decide here.

Reply:
ok

Reply:
My recommendation would be to use bullet points (and to reorder so that plain `re.match` is last, honestly it's a little bit of a footgun)
```
* :func:`re.search` checks for a match anywhere in the string
  (this is what Perl does by default)
* :func:`re.fullmatch` checks for entire string to be a match
* :func:`re.match` checks for a match only at the beginning of the string
```